extensions/amp-bind/0.1/bind-evaluator.js

```diff
@@ -121,15 +122,27 @@ export class BindEvaluator {
         errors[expressionString] = {message: error.message, stack: error.stack};
         return;
       }
+      cache[expressionString] = result;
+    });
+
+    // Then, validate each binding and delete invalid expression results.
+    this.bindings_.forEach(binding => {
+      const {tagName, property, expressionString} = binding;
+      const result = cache[expressionString];
+      if (result === undefined) {
+        return;
+      }
       const resultString = this.stringValueOf_(property, result);
-      if (this.validator_.isResultValid(tagName, property, resultString)) {
-        cache[expressionString] = result;
-      } else {
+      if (!this.validator_.isResultValid(tagName, property, resultString)) {
```

Comment:
I don't think you'll need the delete there, since a result can only get into the cache when isResultValid returns true. 

I'll leave it up to you.